Markdown

agerada opened the pull request:
Implementation of markdown parsing for `acronyms`. 

This required quite extensive code changes, which may have implications for future that will need to be considered. The implementation uses [pandoc lua filters](https://pandoc.org/lua-filters.html#module-pandoc.format) to parse markdown in the short or long names. This seems to capture most markdown characters (at least the common ones that I have tested such as bold, italic, subscript, etc). Existing functionality with case modifiers has been kept, although this is not exhaustive (see text 46).

One of the challenges was the style functions that generate links. These are now more complicated, which is an important limitation. However, compatibility with the extension approach described in `docs/advanced/extending.qmd`) has been maintained.

Markdown parsing is kept as an 'opt in', by default, markdown is not parsed. In the `yaml`, users can declare `markdown:longname` (or `shortname`, or `both`), which will enable parsing. Markdown strings should be wrapped in quotes to ensure that there are no issues in parsing the `yaml`. Markdown can also be triggered (or suppressed) for an individual acronym, see test 42. 

To avoid ambiguous codes, users *must* provide an explicit acronym key when parsing markdown in the shortname. Similiarly, implicit plural is not supported with markdown (plural text must be provided).

GIven the extensive changes proposed in this PR, I will leave as a draft for now until I complete more testing on my documents. Documentation has not been updated (will do once design agreed).

agerada committed:
parse_markdown_in_longname option

agerada committed:
markdown parsers

agerada committed:
apply markdown styles

agerada committed:
tests markdown

agerada committed:
merge links

agerada committed:
explicit plural markdown support

agerada committed:
markdown tests show limitations

agerada committed:
removed redundant parse_simple_italics_markdown

agerada committed:
shortname markdown

agerada committed:
shortname markdown tests

agerada committed:
refactor case conversion

Refactored case conversion to support pandoc inline. Test 44 now fixed.

agerada committed:
test renames

agerada committed:
case test text fixed

agerada committed:
markdown sort patch

agerada committed:
refactor

Allow per acronym markdown setting. Now use markdown: longname|shortname|false
Raise errors when implicit pleural + md attempted.
General refactoring to reduce repeated code and improve readability

agerada committed:
Tests updated with new options

agerada committed:
Backward compatibility with extensions

Restored old create_element function to allow extensions as described in docs/advanced/extending.qmd

agerada committed:
Removed redundant argument from create_rich_element

agerada committed:
refactor out helper functions from styles for clarity

agerada committed:
Test 46 shows case changes limited to common md

rchaput commented:
Woah, you're on a roll!

That was a very complex feature, in my initial tests I found it to conflict with the "custom List of Acronyms" (because it was easier to print acronyms as strings at this point).

It'll take me some time to review it, due to the sheer amount of commits and lines of code; but a few remarks just by reading your description:

- the PR includes the commits for the capitalize option, can you update it to begin with commit [d236e85](https://github.com/rchaput/acronyms/pull/48/commits/d236e8536020f1559d9f59a44638e2573e7b2f41)?
- you mention that markdown parsing is "opt-in", was it required to avoid bugs? I feel like it would be easier if Markdown parsing was on by default. For acronyms that require disabling the Markdown, (for example, `*shortname*` exactly as-is, not interpreted as italic), maybe there is a Pandoc syntax, such as "raw strings" that could do the job?
- I'm not sure it will be useful to allow individual triggering of Markdown, I can't see a case in which one would want to print the acronym without Markdown?
- I agree that users should specify the key when using Markdown in the shortname. It could be difficult to identify the correct acronym otherwise.
- As I said, I remember having problems with the "custom LoA" (meaning using Markdown to specify how the LoA should be rendered), did you try mixing the two? If it does not work, we'll have to mark it as a limitation.

Thanks again for this long awaited feature! I'll try to review as quickly as possible.

agerada committed:
minor refactor

agerada committed:
add make_parenthesized

agerada commented:
Hi @rchaput, thanks for the review! I'm happy to contribute as I have a personal need for this feature. My PhD thesis is otherwise complete and I've commited to use `acronyms` in it. Because I work in a biological domain, the use of italics for bacterial names is not optional for me, so without markdown (or some equivalent), I would probably not be able to use `acronyms`. 

>     * the PR includes the commits for the capitalize option, can you update it to begin with commit [d236e85](https://github.com/rchaput/acronyms/pull/48/commits/d236e8536020f1559d9f59a44638e2573e7b2f41)?

Apologies, I had messed something up with git here (maybe did not sync the branches properly before starting work). Should be fixed now. 

>     * you mention that markdown parsing is "opt-in", was it required to avoid bugs? I feel like it would be easier if Markdown parsing was on by default. For acronyms that require disabling the Markdown, (for example, `*shortname*` exactly as-is, not interpreted as italic), maybe there is a Pandoc syntax, such as "raw strings" that could do the job?
>     * I'm not sure it will be useful to allow individual triggering of Markdown, I can't see a case in which one would want to print the acronym without Markdown?

I agree that it would be easier if markdown was parsed by default. The two main reasons I have not done this, at least for now, is that:

1. The issue with shortname being used as the key. If we always parsed markdown in the shortname then to be safe, users would have to always provide an explicit key. 
2. There may be some unexpected effects of markdown parsing which I have not appreciated yet. By having this as an opt-in feature, the impact of future issues would be minimised. 

Perhaps a reasonable approach would be to parse the longname by default, with the shortname being an opt-in feature. 

Regarding the use of raw/unparsed strings, this is currently possible by escaping the markdown characters (see test 42 for an example). This allows users to mix markdown and non-markdown.

>     * As I said, I remember having problems with the "custom LoA" (meaning using Markdown to specify how the LoA should be rendered), did you try mixing the two? If it does not work, we'll have to mark it as a limitation.

Ah, I didn't appreciate that this was the main issue – I can see why this is tricky. I have tested this, and the good news is that the longname seems to work well. Any markdown added in the custom LoA is simply added to the longname (so for example *E. coli* applied to `**{longname}**` becomes ***E. coli***), which I think would be the desired behaviour. Unfortunately the shortname currently does not work. I think the problem is the addition of the link target as raw markdown. Hopefully I can fix this by applying the link target using pandoc filters instead. 

> Thanks again for this long awaited feature! I'll try to review as quickly as possible.

No problem. I have kept the PR as draft for now, as I plan to do some refactoring for readability to help with the review and future maintenance.

rchaput commented:
Just a quick idea (I need to read your message in more details after work): you mention that you need italics because of your domain. A quick but working option would be to implement a new style, let's say "italics_short" which always puts the shortname in italics. This would be far easier because in Lua we can use pandoc.Italic (or something similar). Thanks to your previous PR, we can even specify the style for each acronym.

I don't mean that it is useless to allow Markdown in acronyms; just that there is a faster way if you (or someone else) require it for your thesis.

I'll try to make some time to review your PR later today.

Sep 18, 2025 11:49:56 Alessandro Gerada ***@***.***>:

> 
>      [Image]*agerada* left a comment (rchaput/acronyms#48)[https://github.com/rchaput/acronyms/pull/48#issuecomment-3306551064]
>     
> 
> Hi @rchaput[https://github.com/rchaput], thanks for the review! I'm happy to contribute as I have a personal need for this feature. My PhD thesis is otherwise complete and I've commited to use *acronyms* in it. Because I work in a biological domain, the use of italics for bacterial names is not optional for me, so without markdown (or some equivalent), I would probably not be able to use *acronyms*.
> 
> ** the PR includes the commits for the capitalize option, can you update it to begin with commit [d236e85](https://github.com/rchaput/acronyms/pull/48/commits/d236e8536020f1559d9f59a44638e2573e7b2f41)?
> *
> Apologies, I had messed something up with git here (maybe did not sync the branches properly before starting work). Should be fixed now.
> 
> ** you mention that markdown parsing is "opt-in", was it required to avoid bugs? I feel like it would be easier if Markdown parsing was on by default. For acronyms that require disabling the Markdown, (for example, `*shortname*` exactly as-is, not interpreted as italic), maybe there is a Pandoc syntax, such as "raw strings" that could do the job?
> * I'm not sure it will be useful to allow individual triggering of Markdown, I can't see a case in which one would want to print the acronym without Markdown?
> *
> I agree that it would be easier if markdown was parsed by default. The two main reasons I have not done this, at least for now, is that:
> 
> 1. The issue with shortname being used as the key. If we always parsed markdown in the shortname then to be safe, users would have to always provide an explicit key.
> 2. There may be some unexpected effects of markdown parsing which I have not appreciated yet. By having this as an opt-in feature, the impact of future issues would be minimised.
> 
> Perhaps a reasonable approach would be to parse the longname by default, with the shortname being an opt-in feature.
> 
> Regarding the use of raw/unparsed strings, this is currently possible by escaping the markdown characters (see test 42 for an example). This allows users to mix markdown and non-markdown.
> 
> ** As I said, I remember having problems with the "custom LoA" (meaning using Markdown to specify how the LoA should be rendered), did you try mixing the two? If it does not work, we'll have to mark it as a limitation.
> *
> Ah, I didn't appreciate that this was the main issue – I can see why this is tricky. I have tested this, and the good news is that the longname seems to work well. Any markdown added in the custom LoA is simply added to the longname (so for example /E. coli/ applied to ***{longname}*** becomes /*E. coli*/), which I think would be the desired behaviour. Unfortunately the shortname currently does not work. I think the problem is the addition of the link target as raw markdown. Hopefully I can fix this by applying the link target using pandoc filters instead.
> 
> Thanks again for this long awaited feature! I'll try to review as quickly as possible.
> 
> No problem. I have kept the PR as draft for now, as I plan to do some refactoring for readability to help with the review and future maintenance.
> 
> —
> Reply to this email directly, view it on GitHub[https://github.com/rchaput/acronyms/pull/48#issuecomment-3306551064], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AGZR5IIU4KQJ2PHRCYRJM2D3TJ54JAVCNFSM6AAAAACGPVS2A2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTGMBWGU2TCMBWGQ].
> You are receiving this because you were mentioned.
> [Tracking image][https://github.com/notifications/beacon/AGZR5INDXEQ55PPJCYXIEK33TJ54JA5CNFSM6AAAAACGPVS2A2WGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTWFCX3RQ.gif]
>

agerada commented:
That's an interesting thought that I hadn't considered. Unfortunately, many of the acronyms are mixtures of different cases, for example methicillin-resistant _Staphylococcus aureus_ (MRSA), or _Clostridium difficile_ infection (CDI). Sadly I don't think it will work (unless using very complex styles, and then it might as well be markdown). Fortunately the thesis is rendering great with `acronyms` installed from this branch, including italics where required.

rchaput commented:
I've had time to read your comments in details

> I agree that it would be easier if markdown was parsed by default. The two main reasons I have not done this, at least for now, is that:
> 
>     1. The issue with shortname being used as the key. If we always parsed markdown in the shortname then to be safe, users would have to always provide an explicit key.

I'm wondering whether we can use Pandoc to detect markdown in the shortname. If we could do that, we could: 1) detect if there is markdown; 2) if there is, check if the key is defined, and raise an error otherwise. I don't remember the notions of `raw_strings` and others like that enough...

>     2. There may be some unexpected effects of markdown parsing which I have not appreciated yet. By having this as an opt-in feature, the impact of future issues would be minimised.

I agree that it can be risky. Then again, if we have some kind of feature to disable Markdown parsing (I'm thinking something like `shortname: r"This is *not* Markdown"`), we can mitigate the impact as well, while still simplifying use. I'm not sure many acronyms will include Markdown-specific characters.

> Perhaps a reasonable approach would be to parse the longname by default, with the shortname being an opt-in feature.

Do you think we could enable Markdown shortname if and only if the key is set? That would avoid including an additional `markdown: shortname` option, but I fear that this would confuse people. From a technical point of view, it makes sense, since we require an explicit key to avoid ambiguous shortname->key conversion; but this may not be clear...

> Regarding the use of raw/unparsed strings, this is currently possible by escaping the markdown characters (see test 42 for an example). This allows users to mix markdown and non-markdown.

I've seen the example, it's good that we can escape some characters, but I was thinking of escaping the whole string. As I mentioned earlier, I remember some specific syntax like `r"raw string"`, but maybe I'm confusing with Python, I'm not sure...

> Ah, I didn't appreciate that this was the main issue – I can see why this is tricky. I have tested this, and the good news is that the longname seems to work well. Any markdown added in the custom LoA is simply added to the longname (so for example _E. coli_ applied to `**{longname}**` becomes _**E. coli**_), which I think would be the desired behaviour. Unfortunately the shortname currently does not work. I think the problem is the addition of the link target as raw markdown. Hopefully I can fix this by applying the link target using pandoc filters instead.

It's great that the longname already works. If we can make the shortname work as well, it's best; but I'm OK with stating it as a limitation if it doesn't. I think that allowing Markdown is too much important.

agerada committed:
Patch to fix custom loa with markdown

A new helper function (serialize_inlines) added, with collapses pandoc inlines into a string so that they can be parsed again for the custom loa. Tests added. Supports markdown in both shortname and longname.

agerada commented:
I think I have managed to fix the custom LoA (https://github.com/rchaput/acronyms/pull/48/commits/a34f1be0d9accadd71f069f9cf20cd4b602c4c3c). Markdown is now added to the shortname as well (even when using markdown in the custom LoA), and the links are working as expected.

> I'm wondering whether we can use Pandoc to detect markdown in the shortname. If we could do that, we could: 1) detect if there is markdown; 2) if there is, check if the key is defined, and raise an error otherwise. I don't remember the notions of `raw_strings` and others like that enough...

This would solve a lot of problems as I think it would then be reasonabe to enable markdown by default (and have an option to suppress or use raw strings). If markdown is detected in the shortcode in the absence of an explicit key, then an error can be raised, asking the user to provide a key. That would solve most of the issues. The only difficulty is whether there is a way to reliably look for markdown in the parsed Pandoc (without having to list out all the possible blocks such as Emph, etc; which would be prone to breaking). I'll look into this.

agerada committed:
Markdown parsed by default

Both shortname and longname now parse md by default.
This needed a new helper (contains_markdown), that allows us to check for markdown in an input. This is mainly used to check for markdown in the shortname, and trigger errors if true and no explicit key. It's also used to trigger an error if markdown + attempt to use plural without explicit plural term.
This has needed some updates to styles (mainly to fix footnotes, which had not been compatible with markdown).

Finally, options (parse_markdown_*) are still used, but set to default true. These can be removed altogether once feature added to support raw strings.

agerada committed:
Remove unnecessary element creation function

As we can now assume input is inlines, legacy create_element is not required, as create_rich_element replaces it. To keep compatibility with extending.qmd, kept local reference to create_element.

agerada committed:
Refactor styles

General refactor and clean up to styles functions.

agerada committed:
Remove optional markdown code

This commit removes the internal features that allowed markdown to be optional. Now markdown will always be parsed. Characters can still be escaped if necessary (see test 42).

agerada commented:
After giving this some thought. I fully agree that markdown should be parsed by default. I have enabled this.

I have also removed the internal workings that allowed markdown to be turned on or off (https://github.com/rchaput/acronyms/pull/48/commits/a405c2a1f7e6d8f2de8040b756c16d3a56a3278f). This has allowed me to clean up the code and remove a lot of checking to see whether markdown is being parsed. There is now a `contains_markdown` function that should detect markdown and ask the user to provide a `key` (if shortname has markdown).

The issue of escaping markdown-like characters (which, to be fair, should be rarely needed), I think is sufficiently covered by escaping using `\\`. I imagine it would be possible to have a raw string (or equivalent option), but it would make the internal workings more complicated (needing to check if raw string or inline), for something that already has a workaround. The only annoyance is that two `\\` are needed, so that would have to be documented. 

Hopefully the code is cleaner now, so I have removed the draft status of this PR.

rchaput commented on _extensions/acronyms/acronyms.lua:
```diff
@@ -65,21 +66,42 @@ function Acronym:new(object)
+    -- Enforce explicit key when markdown parsing for shortname is enabled.
+    if Helpers.contains_markdown(object.shortname) then
+        if object.key == nil then
+            quarto.log.error("[acronyms] Each acronym must provide an explicit `key` when using markdown in shortname.")
```
It would be useful to print the acronym shortname here so it's clear for the user which one to fix.

agerada replied:
Added 82930b41e35adce853d8befa97729ed74181ffa8

rchaput commented on _extensions/acronyms/acronyms.lua:
```diff
@@ -65,21 +66,42 @@ function Acronym:new(object)
+    if object.plural then
+        if object.plural.shortname ~= nil then explicit_plural_short = true end
+        if object.plural.longname  ~= nil then explicit_plural_long  = true end
+    end
```
We can simplify this (easier to read)

```suggestion
    local explicit_plural_short = object.plural and object.plural.shortname ~= nil
    local explicit_plural_long  = object.plural and object.plural.longname ~= nil
```

rchaput commented on _extensions/acronyms/acronyms.lua:
```diff
@@ -218,12 +247,16 @@ function Acronyms:parseFromMetadata(metadata, on_duplicate)
+        -- Always parse markdown for names
+        local shortname = Helpers.extract_meta_field(v.shortname, true)
+        local longname  = Helpers.extract_meta_field(v.longname,  true)
+    -- raw longname parsed
```
I don't understand this comment?

agerada replied:
old comment, sorry aab2c99bfe71096ca5c5f1c4121177223e2528f5

rchaput commented on _extensions/acronyms/acronyms_helpers.lua:
```diff
@@ -104,5 +104,272 @@ function Helpers.str_to_boolean(value)
+    if value == nil then return false end
+
+    -- If we have a Pandoc type, check AST nodes directly
+    if pandoc and pandoc.utils and pandoc.utils.type then
```
Why checking for `pandoc.utils.type`? Is there a Pandoc version in which we do not have access to this function? We can use the `Helpers.is_at_least(version)` for this kind of problem.

agerada replied:
Agreed, see 62a1c1e05a4614911eb6c9006b2c958fcf3d13cd

rchaput commented on _extensions/acronyms/acronyms_helpers.lua:
```diff
@@ -104,5 +104,272 @@ function Helpers.str_to_boolean(value)
+            if type(v) == "table" and v.t ~= nil then
+                if v.t ~= "Str" and v.t ~= "Space" and v.t ~= "SoftBreak" and v.t ~= "LineBreak" then
+                    return true
+                end
```
This condition is the same as in the previous loop (case). Can we merge the two loops? (in order to simplify the code, not making it harder to read)

agerada replied:
Agreed, have changed here: 943a3aff1c3e9657db54e21cf5dce74e3abc256c

rchaput commented on _extensions/acronyms/acronyms_helpers.lua:
```diff
@@ -104,5 +104,272 @@ function Helpers.str_to_boolean(value)
+                if v.t ~= "Str" and v.t ~= "Space" and v.t ~= "SoftBreak" and v.t ~= "LineBreak" then
+                    return true
+                end
+            end
+        end
+    end
```
This case does not return `false`, contrary to the previous case; is there a reason for that?
(note that this joins my previous comment; if both cases should return false, it would be clearer to merge them)

agerada replied:
I think this was a bug which just wasn't captured in testing. Now using a single function for this (see above).

rchaput commented on _extensions/acronyms/acronyms_helpers.lua:
```diff
@@ -104,5 +104,272 @@ function Helpers.str_to_boolean(value)
+       or s:find("_.-_")      -- _emph_
+       or s:find("`")         -- inline code
+       or s:find("%[.-%]%(")  -- [text](link)
+       or s:find("!%[.-%]%(") -- image
```
Image is very similar to `[text](link)` and should be detected by the same regex I guess.
However, the strike-through Markdown (`~~strike this~~`) is not detected; we should add this

agerada replied:
Modified regex slightly to capture both image and link in one. Added strike through. 943a3aff1c3e9657db54e21cf5dce74e3abc256c

rchaput commented on _extensions/acronyms/acronyms_helpers.lua:
```diff
@@ -104,5 +104,272 @@ function Helpers.str_to_boolean(value)
+        for _, il in ipairs(src) do
+            if il.t == "Str" and (il.text or il.c) then
+                local txt = il.text or il.c
+                if case_kind == "upper" then
+                    txt = txt:upper()
+                elseif case_kind == "lower" then
+                    txt = txt:lower()
+                elseif case_kind == "sentence" and not done_first then
+                    local i = txt:find("%a")
+                    if i then
+                        txt = txt:sub(1,i-1)..txt:sub(i,i):upper()..txt:sub(i+1)
+                        -- set done_first to ensure only one capitalization in sentence case
+                        done_first = true
+                    end
+                end
+                dest[#dest+1] = pandoc.Str(txt)
+            elseif simple_containers[il.t] and type(il.c) == "table" then
+                local inner = transform_inlines(il.c)
+                if il.t == "Span" then
+                    dest[#dest+1] = pandoc.Span(inner, il.attr)
+                else
+                    local ctor = pandoc[il.t]
+                    if ctor then
+                        dest[#dest+1] = ctor(inner)
+                    else
+                        local copy = {}
+                        for k, v in pairs(il) do copy[k] = v end
+                        copy.c = inner
+                        dest[#dest+1] = copy
+                    end
+                end
+            else
+                dest[#dest+1] = il
+            end
+        end
+        return dest
+    end
+
+    return transform_inlines(value)
+end
```
Is it necessary to put this function here in Helpers? My idea for Helpers was to put here functions that can be used in several places. I think that `transform_case` is only used when rendering an acronym, so maybe we could place this in `acronyms_styles`? (or perhaps `acronyms_pandoc`?)

agerada replied:
Fair enough. I have also moved `make_parenthesized` and `create_rich_element` to `acronyms_styles` as well, as they will only be used in that file. 5c5ed3f52639846c8c900d1bed44031db6565f61

rchaput commented on _extensions/acronyms/acronyms_pandoc.lua:
```diff
@@ -133,14 +133,13 @@ end
 function AcronymsPandoc.generateDefinitionList(sorted_acronyms)
     local definition_list = {}
     for _, acronym in ipairs(sorted_acronyms) do
-        -- The definition's name. A Span with an ID so we can create a link.
```
Why delete this comment? Is it no longer up-to-date with the code?

agerada replied:
I think I removed this while modifying this function, but have returned here: 44b114981e2aa9d4037ddfeffd6cda2a088ef37c

rchaput commented on _extensions/acronyms/acronyms_pandoc.lua:
```diff
@@ -133,14 +133,13 @@ end
-        -- The definition's value.
-        local definition = pandoc.Plain(acronym.longname)
-        table.insert(definition_list, { name, definition })
+        local long_inlines = Helpers.ensure_inlines(acronym.longname)
+        local longname_val = pandoc.Plain(long_inlines)
+        table.insert(definition_list, { name, longname_val })
```
```suggestion
        local definition_value = pandoc.Plain(Helpers.ensure_inlines(acronym.longname))
        table.insert(definition_list, { name, definition_value })
```

rchaput commented on _extensions/acronyms/acronyms_styles.lua:
```diff
@@ -35,82 +35,81 @@
 -- Local helper function to create either a Str or a Link,
 -- depending on whether we want to insert links.
-local function create_element(content, key, insert_links)
+-- Kept as a local reference here to keep compatibility with previous versions.
+create_element = Helpers.create_rich_element
```
I'm not sure we need to keep this; we can simply replace the reference in all styles, and update the documentation.

rchaput commented on _extensions/acronyms/acronyms_styles.lua:
```diff
@@ -139,33 +138,34 @@ return function(acronym, style_name, insert_links, is_first_use, plural,
+        if need_short_strict then
+            quarto.log.error("[acronyms] Plural form requested for '" .. tostring(acronym.key) .. "' but 'plural.shortname' was not explicitly provided while markdown parsing is enabled for its shortname. Define it under plural: { shortname: ... } to use \\acrs{" .. tostring(acronym.key) .. "} .")
+            assert(false)
+        end
```
I would prefer raising the error before, typically in `acronyms_pandoc` (if I remember correctly, the function should be something like `replaceExistingAcronym`). This is simply checking that an option is correctly set.
The `style` function should perform very minimal error-checking (such as checking the acronym is not nil, otherwise the function cannot work!).

agerada replied:
Agree, much better to raise earlier (and still has the advantage of only raising if plural was attempted, so no unecessary errors). See 66a976a452ba7bae3dec1a5b26a284822ca0e9e9. I have tested this informally as the current testing suite does not (as far as I can see) support expected exceptions.

rchaput commented on _extensions/acronyms/acronyms_styles.lua:
```diff
@@ -139,33 +138,34 @@ return function(acronym, style_name, insert_links, is_first_use, plural,
-            acronym.longname = capitalize_first(acronym.longname)
-        end
+    -- Delegate case transformation to Helpers.transform_case which preserves inline formatting
+    local function transform_case(value, case_kind)
+        return Helpers.transform_case(value, case_kind)
+    end
+
+    if case_target == "short" or case_target == "both" then
+        acronym.shortname = transform_case(acronym.shortname, case)
+    end
+    if case_target == "long" or case_target == "both" then
+        acronym.longname = transform_case(acronym.longname, case)
     end
```
Why use a local function that delegates to `Helpers`?

```suggestion
    if case_target == "short" or case_target == "both" then
        acronym.shortname = Helpers.transform_case(acronym.shortname, case)
    end
    if case_target == "long" or case_target == "both" then
        acronym.longname = Helpers.transform_case(acronym.longname, case)
    end
```

agerada replied:
Oops, this was a silly one!

rchaput reviewed (commented):
I've read most of your changes; lots of great ideas!

I think it needs a bit of refactoring to simplify the code, but everything should work well (just as you tested)

agerada committed:
make_parenthesized to helpers

agerada committed:
line width

agerada committed:
md docs

agerada committed:
print acronyms with missing key

agerada committed:
Apply suggestion from @rchaput

Co-authored-by: Remy Chaput <rchaput.pro@gmail.com>

agerada committed:
clean up old comments from acronyms.lua

agerada committed:
check pandoc version when using type

agerada committed:
simplify contains_markdown

agerada committed:
move transform_case create_element and make_parenthesized to acronym_styles

agerada committed:
Apply suggestion from @rchaput

Co-authored-by: Remy Chaput <rchaput.pro@gmail.com>

agerada committed:
Reinsert definition list comments

agerada committed:
move plural error raising fro styles to acronyms_pandoc

agerada committed:
Apply suggestion from @rchaput

Co-authored-by: Remy Chaput <rchaput.pro@gmail.com>

agerada reviewed (commented):
Thanks for the review; I appreciate there was a lot of code to go through here! Hopefully all comments addressed but happy to look at anything else that comes up.

Commit:
Replace master branch with main

The default branch was named `master`, but Quarto recommends using
`main`. In addition, the docs was not consistent, and sometimes used an
unworking command.
This commit changes the few documents that refer to the `master` branch
(including the CI/CD and direct links to files in the repository).

Commit:
Add instructions to install specific version

psomhorst commented:
Does this feature also support math (e.g., a double subscript `$P_{E_1}$` => $P_{E_1}$) in shortnames or longnames?

agerada commented:
Hi @psomhorst. The current implementation almost certainly does not support math, although I haven't specifically tested this. I'm not sure how feasible math support would be, it mainly depends on whether there is a Pandoc lua filter to parse it. I can look into this.

agerada committed:
Merge branch 'rchaput:main' into markdown

agerada committed:
Test maths in names

agerada commented:
Good news, it seems like I was wrong! Math <ins> is</ins> being parsed in this feature. I didn't need to change any of the code, but added a couple of tests (https://github.com/rchaput/acronyms/pull/48/commits/f1c3cffe692ff978c819a02426d9026899ebb627) to demonstrate it. Although I haven't exhaustively tested edge cases, I didn't run into any issues while informally testing it out. Perhaps good for this to be covered in the documentation as well, as it might be important for some scientific domains.

psomhorst commented:
Fantastic! If you want, I can supply some cases or documentation from a scientific writing standpoint.

agerada commented:
If you do have a relevant example or two from your domain that demonstrate the use of mathematics in acronyms then I think it would be quite helpful.

Commit:
Suggestions for examples of acronyms used in scientific writing

psomhorst commented:
> If you do have a relevant example or two from your domain that demonstrate the use of mathematics in acronyms then I think it would be quite helpful.

See [this PR](https://github.com/agerada/acronyms/pull/3).

agerada committed:
Merge pull request #3 from psomhorst/suggestions-examples

Suggestions for examples of acronyms used in scientific writing
